ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/query/LogsQuery.java

```diff
@@ -19,20 +19,54 @@
 import org.hyperledger.besu.ethereum.core.Address;
 import org.hyperledger.besu.ethereum.core.Log;
 import org.hyperledger.besu.ethereum.core.LogTopic;
+import org.hyperledger.besu.ethereum.core.LogsBloomFilter;
 
 import java.util.Arrays;
 import java.util.List;
+import java.util.Objects;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
 
+import com.google.common.base.Suppliers;
 import com.google.common.collect.Lists;
 
 public class LogsQuery {
 
   private final List<Address> queryAddresses;
   private final List<List<LogTopic>> queryTopics;
+  private final Supplier<List<LogsBloomFilter>> addressBlooms;
+  private final Supplier<List<List<LogsBloomFilter>>> topicsBlooms;
+
+  private LogsQuery(final List<Address> queryAddresses, final List<List<LogTopic>> queryTopics) {
+    this.queryAddresses = queryAddresses;
+    this.queryTopics = queryTopics;
+    addressBlooms =
+        Suppliers.memoize(
```

Comment:
Are we expecting to make a lot of `LogsQuery`s that we won't use? If not, Maybe we should just do the bloom filter computation up front.